.ci/merge-fixes.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+# Merge open PRs from sagemath/sage labeled "blocker".
+REPO="sagemath/sage"
+GH="gh -R $REPO"
+PRs="$($GH pr list --label "p: blocker / 1" --json number --jq '.[].number')"
```

Comment:
You probably want to make sure that they also have the "postive review" label.

Reply:
Definitely not. Sometimes it takes a while to find a competent reviewer.

Reply:
See https://github.com/sagemath/sage/pull/36338#issuecomment-1736131315 for @mkoeppe's opinion on this.

Without requiring "positive review" label, we can check the PR before we set it positive review.

Reply:
It will influence every other PR, so I think there should be some safety measure.

> Without requiring "positive review" label, we can check the PR before we set it positive review.

What do you mean? Doesn't the CI of that PR already checks the PR? Why do you need other PRs to check it?

Reply:
> so I think there should be some safety measure.

We already have that, namely the Triage group + common sense.

Reply:
> > Without requiring "positive review" label, we can check the PR before we set it positive review.
> 
> What do you mean? Doesn't the CI of that PR already checks the PR? 

Perhaps I am confused and you are right.

Then I am inclined to agree with you. People can make mistakes. Another safety measure does not hurt.

Reply:
Requiring a positive review does hurt because it slows down the response.

Reply:
Still better in my opinion than that the combo "blocker + needs review" would mean "everyone now has to cope with my experimental changes that nobody looked at yet". "blocker" is priority label (i.e. meaning: please have a look at this asap) and nothing more.

If you want to make the process easier for everyone, just prevent that PRs are merged without them passing the CI...

Reply:
In actual practice observed in our repo, people here simple do not use the "blocker" label frivolously. We can trust our developers.

And of course I have considered other possibilities before making the decision on "blocker", regardless of review status; in particular, introducing a separate label just for this task.

Reply:
Let's please just keep it simple. A practical problem, a simple solution.

Reply:
> Requiring a positive review does hurt because it slows down the response.

We are not putting off fire. There's no need to act immediately as soon as someone pushes a CI-fix PR and adds blocker label.  We all want to fix CI failures as soon as possible and it would not take long before another one would lend second eyes to check the PR and give positive review.

But, as the author of this PR wants it that way, please @tobiasdiez waive your (and my) worries about broken Ci-fix PRs. If the worries turns out to be real, then we can fix it then (I guess @mkoeppe is the first to fix it :-)

Reply:
Sorry, but I don't understand the argument here: because two people don't like the change, we go along with the solution that one person prefers?

I actually think this PR here is a perfect example: there is absolutely no reason this is tagged as a blocker, it doesn't resolve any issues with falling CIs or something similar, it is not tested very well (or did any of the reviewers tried to run the new macos workflow on the main repo?) and there are open discussions about how to implement certain things without too much disadvantages (it's another question how serious these points are). Why should this PR be merged into every other PR?

I'm happy with either of the following solutions:
- Introduce a new label that triggers the auto-merging (that is then assigned once someone looked over the changes and gave them a thumps up, even if there might be some smaller issues still outstanding)
- Auto-merge all positively reviewed PRs (that would be then a poor-mans merge queue)
- Only merge with positive review + blocker.

Reply:
> Sorry, but I don't understand the argument here: because two people don't like the change, we go along with the solution that one person prefers?

I don't count, I weigh, and I weigh more the opinion of the author (who did most of the work).

Reply:
> We are not putting off fire. There's no need to act immediately as soon as someone pushes a CI-fix PR and adds blocker label. 

Well, it's not fire, but it's a matter of "availability of the CI services", which, if poor, leads to developer frustration (@chapoton @tornaria) and can dramatically reduce development velocity because developers may disengage or wait until the CI is fixed again.

Let's consider the period from the release of a beta version to the release of the next beta version, currently about a week. For, say, 95% of availability of the CI, we want a CI fix to take effect within a few hours of a broken release.

> We all want to fix CI failures as soon as possible and it would not take long before another one would lend second eyes to check the PR and give positive review.

This may be true for trivial fixes such as the linter failures, which can be fixed and reviewed by many people.

But for fixes to CI and build system, it hinges on the availability of a very small number of people, and time-to-positive-review can be several days -- which can easily make the availability of the CI services less than 50%.

Reply:
@tobiasdiez wrote:
> Sorry, but I don't understand the argument here: because two people don't like the change, we go along with the solution that one person prefers?

I know that @tobiasdiez knows this already, but I'll point out again: Our project does not hold majority votes In PR discussions.

> I actually think this PR here is a perfect example: there is absolutely no reason this is tagged as a blocker, it doesn't resolve any issues with falling CIs or something similar, 

This is false. This PR does resolve the issue of the failing CIs. Merging this PR is enough to fix all CI problems in the current cycle: it auto-applies the blocker PRs #36327 and #36276.

> it is not tested very well (or did any of the reviewers tried to run the new macos workflow and the other changed workflows on the main repo?)

This is false, and moreover an inappropriate attack on author and reviewers.

Reply:
-1 on holding majority votes in PRs.

Reply:
> @tobiasdiez wrote:
> 
> > Sorry, but I don't understand the argument here: because two people don't like the change, we go along with the solution that one person prefers?
> 
> I know that @tobiasdiez knows this already, but I'll point out again: Our project does not hold majority votes In PR discussions.
> 
> > I actually think this PR here is a perfect example: there is absolutely no reason this is tagged as a blocker, it doesn't resolve any issues with falling CIs or something similar,
> 
> This is false. This PR does resolve the issue of the failing CIs. Merging this PR is enough to fix all CI problems in the current cycle: it auto-applies the blocker PRs #36327 and #36276.
> 
> > it is not tested very well (or did any of the reviewers tried to run the new macos workflow and the other changed workflows on the main repo?)
> 
> This is false, and moreover an inappropriate attack on author and reviewers.

Sadly not false...